impeller/aiks/paint.cc

```diff
@@ -99,6 +99,11 @@ std::shared_ptr<Contents> Paint::WithImageFilter(
 std::shared_ptr<Contents> Paint::WithColorFilter(
     std::shared_ptr<Contents> input,
     bool absorb_opacity) const {
+  // Image input types will directly set their color filter,
+  // if any. See `TiledTextureContents.SetColorFilter`.
+  if (color_source_type == ColorSourceType::kImage) {
+    return input;
```

Comment:
Won't this will break if the filter is set after the color source?

Reply:
Oh, it's capturing a ref to the paint.

Reply:
We can lose this optimization if its making canvas weird. now that we're handling filters better in general its not as necessary.

Reply:
We can lose this optimization if its making canvas weird. now that we're handling filters better in general its not as necessary.

Reply:
Nah I think it's alright. Just had to rethink a refactor I'm working on.